### PR TITLE
Disable iOS ARM validation

### PR DESCRIPTION
**Pull request by lukas-lansky**
As discussed in https://github.com/dotnet/runtime/pull/50940

Also, backports test fix from https://github.com/dotnet/runtime/pull/48261

### CODE REVIEW

**Review by mmitche — APPROVED** [review]

**Comment by akoeplinger** [comment]
Looks good but I think we should disable all of iOS/tvOS/Android builds in 5.0 since we ended up not shipping them there.

**Comment by mmitche** [comment]
@akoeplinger can you handle that on this PR to unblock 5.0?

**Review by akoeplinger — APPROVED** [review]

**Comment by lukas-lansky** [comment]
@akoeplinger @mmitche I don't have merge rights here, so please feel free to finish the PR at your convenience.

**Pull request merged**
